src/test/java/de/tum/in/www1/artemis/JhiMetricsIntegrationTest.java

```diff
@@ -0,0 +1,45 @@
+package de.tum.in.www1.artemis;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+class JhiMetricsIntegrationTest extends AbstractSpringIntegrationIndependentTest {
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Test
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    public void getMetricsTest() throws Exception {
+        var result = request.get("/management/jhimetrics", HttpStatus.OK, String.class);
+
+        JsonNode rootNode = objectMapper.readTree(result);
+
+        assertThat(rootNode.has("jvm")).isTrue();
+        assertThat(rootNode.path("jvm").has("G1 Old Gen")).isTrue();
+        assertThat(rootNode.path("jvm").path("G1 Old Gen").has("committed")).isTrue();
+        assertThat(rootNode.path("jvm").path("G1 Old Gen").has("max")).isTrue();
+        assertThat(rootNode.path("jvm").path("G1 Old Gen").has("used")).isTrue();
+
+        assertThat(rootNode.has("http.server.requests")).isTrue();
+        assertThat(rootNode.path("http.server.requests").path("all").has("count")).isTrue();
+
+        assertThat(rootNode.has("cache")).isTrue();
+
+        assertThat(rootNode.has("garbageCollector")).isTrue();
+        assertThat(rootNode.path("garbageCollector").has("jvm.gc.max.data.size")).isTrue();
+        assertThat(rootNode.path("garbageCollector").path("jvm.gc.pause").has("max")).isTrue();
```

Comment:
Hmm not sure tbh. if it is a configuration issue or not.

I can't reproduce this locally, as I get it back like this:

```
"garbageCollector" : {
    "jvm.gc.max.data.size" : 3.221225472E9,
    "jvm.gc.pause" : {
      "max" : 32.0,
      "totalTime" : 32.0,
      "mean" : 32.0,
      "count" : 1
    },
    "jvm.gc.memory.promoted" : 4505632.0,
    "jvm.gc.memory.allocated" : 5.95591168E8,
    "classesLoaded" : 36184.0,
    "jvm.gc.live.data.size" : 0.0,
    "jvm.gc.overhead" : 0.003392783629755711,
    "classesUnloaded" : 4.0
  },
```
But from my debugging and testing I think I remember that once some parts of the garbage collection metrics didn't show up initially, but then were shown after a reload. Not sure if that's related though.

Have you repeated running the test, and it keeps failing? If that is the case it might be a configuration issue. You could try to invalidate the caches in intellij.

Reply:
The Test also works perfectly on my machine, just FYI 👍

Reply:
I tried on another computer and it works without any problems. I think it‘s just something weird with my IDE. 😅